### ISSUE TITLE
Bun install on Ubuntu 21.10 gives SementationFault at 188 error

### ISSUE THREAD

**Issue by kelarcrisp** [oven-sh/bun#273]
Seeing this error when trying to install dependencies from a package.json file

![image](https://user-images.githubusercontent.com/49597486/177592269-1494ff1e-5a3c-484f-9f9c-47a5afe4675a.png)

**Comment by pierrick-martin-dev** [comment]
Have you a specific config ?

**Comment by kelarcrisp** [comment]
@flowHater nope

**Comment by pierrick-martin-dev** [comment]
Have you some package that come from an other registry that the default one ?

**Comment by Jarred-Sumner** [comment]
@kelarcrisp Can you paste the package.json?

**Comment by kelarcrisp** [comment]
@Jarred-Sumner as requested 

```js
  "dependencies": {
    "@date-io/date-fns": "^2.11.0",
    "@emotion/cache": "^11.7.1",
    "@emotion/react": "^11.7.1",
    "@emotion/server": "^11.4.0",
    "@emotion/styled": "^11.6.0",
    "@mui/icons-material": "^5.2.1",
    "@mui/lab": "^5.0.0-alpha.62",
    "@mui/material": "^5.6.3",
    "@sentry/nextjs": "^6.19.7",
    "@sentry/webpack-plugin": "^1.18.9",
    "axios": "^0.24.0",
    "babel-plugin-module-resolver": "^4.1.0",
    "date-fns": "^2.28.0",
    "formik": "^2.2.9",
    "jest-environment-jsdom": "^28.0.2",
    "js-cookie": "^3.0.1",
    "mobx": "^6.3.10",
    "mobx-react": "^7.2.1",
    "next": "^12.1.5",
    "next-compose-plugins": "^2.2.1",
    "next-images": "^1.8.4",
    "next-svgr": "^0.0.2",
    "prop-types": "^15.7.2",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-error-boundary": "^3.1.4",
    "react-query": "^3.34.6",
    "window-or-global": "^1.0.1",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@testing-library/dom": "^8.11.1",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^13.1.1",
    "@testing-library/react-hooks": "^8.0.0",
    "@testing-library/user-event": "^14.1.1",
    "be-pretty": "0.9.5",
    "css-mediaquery": "^0.1.2",
    "eslint": "^8.4.1",
    "eslint-config-airbnb": "^19.0.2",
    "eslint-config-next": "12.0.7",
    "eslint-config-prettier": "^8.3.0",
    "eslint-import-resolver-babel-module": "^5.3.1",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-jest": "^26.1.5",
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.27.1",
    "eslint-plugin-react-hooks": "^4.3.0",
    "eslint-plugin-testing-library": "^5.0.1",
    "jest": "^28.0.3",
    "jest-axe": "^6.0.0",
    "next-router-mock": "^0.6.7",
    "prettier": "^2.5.1",
    "pretty-quick": "^3.1.2",
    "react-axe": "^3.5.4",
    "react-test-renderer": "^18.1.0"
  }
}

```

**Comment by charafsalmi** [comment]
I have the very same issue, but under windows. The machine is behind a proxy. It may also be related to SSL/TLS misconfiguration.
```
GET /@types/serve-index
	[0m[36mIf-None-Match[0m[2m: [0m"9ad5f009b7a096fe1e7f7a00d74863b0"
	[0m[36mAccept[0m[2m: [0mapplication/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*
	[0m[36mConnection[0m[2m: [0mclose
	[0m[36mUser-Agent[0m[2m: [0mbun.js 0.1.4
	[0m[36mHost[0m[2m: [0mregistry.npmjs.org
	[0m[36mAccept-Encoding[0m[2m: [0mdeflate, gzip
SegmentationFault at 188
```

**Comment by Jarred-Sumner** [comment]
Should be fixed in Bun v0.1.12
